File: 3163-string-compression-iii/3163-string-compression-iii.py
class Solution:
    def compressedString(self, word: str) -> str:
        result = []
        count = 1
        word += "."
        prev = word[0]
        for i in range(1, len(word)):
            curr = word[i]
            if curr == prev and count < 9:
                count += 1
                continue
            result.append(f"{count}{prev}")
            count = 1
            prev = curr
        return "".join(result)

File: 3163-string-compression-iii/3163-string-compression-iii.cpp
class Solution {
public:
    string compressedString(string word) {
        word.push_back('.');

        string result;
        char prev = word[0];
        int count = 1;
        int n = word.size();

        for (int i = 1; i < n; ++i) {
            char curr = word[i];

            if (curr == prev && count < 9) {
                count += 1;
            } else {
                result.push_back('0' + count);
                result.push_back(prev);
                prev = curr;
                count = 1;
            }
        }
        return result;
    }
};
